pkg/cache/cache.go

```diff
@@ -249,6 +249,14 @@ type ByObject struct {
 	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
 	// otherwise you will mutate the object in the cache.
 	UnsafeDisableDeepCopy *bool
+
+	// NewCache allows to use a custom cache creation function for the object.
+	// The Options object passed to the NewCache function is pruned to only contain
+	// the object's relevant ByObject configuration.
+	//
+	// NOTE: LOW LEVEL PRIMITIVE!
+	// Only use a custom NewCache if you know what you are doing.
+	NewCache NewCacheFunc
```

Comment:
Sorry, but no. The `NewCache` func gets the cache.Options already. You can then inside your custom `NewCache` func decide what to do on a per-object basis.

/hold

Reply:
You have a point here. 
It would be nice to combine both, the controller-runtime's cache implementation (to be used as a default cache) and a custom implementation on a per object basis (via `delegatingByGVKCache`). 
Is there any acceptable way to achieve this?

Reply:
Would you be open to have any custom cache function on an per object basis at all?
It's definitely a use-case for us and we'd be happy to reuse the `delegatingByGVKCache` implementation. I guess making this implementation public isn't an option either?!

Please let me know what you think.

Reply:
What we try to accomplish is to use a custom cache implementation for some kinds, while falling back to the general cache for others.

Specifically, we use a single-object cache implementation due to security restrictions. It's similar to `kubelet` which also uses such a cache ([ref](https://github.com/kubernetes/kubernetes/blob/v1.26.0/pkg/kubelet/util/manager/watch_based_manager.go)) to play well with the [NodeRestriction](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction) admission plugin.

Reply:
That should work well with what @alvaroaleman suggested? Seems like a layer on top, of this implementation, rather than something controller runtime should handle

Reply:
> It's definitely a use-case for us and we'd be happy to reuse the delegatingByGVKCache implementati

Copy it? The issue I have with making it public or adding the option you suggest here is that sooner or later someone will come up with a use-case where neither of the two fits and they want to have another layer at which they want to configure things.

IMHO what we currently have covers 98% of use-cases and leaves the low-level escape hatch of using the `NewCacheFunc`. Our goal IMHO is not to make 100% of use-cases easy because the more esoteric ones will actually make the more standard one confusing but to make the majority of use-cases easy and all possible, which I think we do through exposing the `NewCache` func.

Reply:
Sure, it works with our implementation or if we copy the code. Still thought it was worth a shot to contribute our use-case back to the community. Thanks for your feedback 🙂